specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-01-01/frontdoor.json

```diff
@@ -1237,12 +1237,28 @@
           "description": "Location of the backend (IP address or FQDN)",
           "type": "string"
         },
-        "privateLinkId" : {
-          "description": "If this backend is private, provide the private link resource Id. Populating this optional field indicates that this backend is 'Private'",
+        "privateLinkResourceId": {
+          "description": "If this backend is private, provide the Private Link resource Id. Populating this optional field indicates that this backend is 'Private'",
           "type": "string"
         },
-        "location": {
-          "description": "Region of the backend if it is in Azure. Mandatory field if backend is 'Private'",
+        "privateEndpointStatus": {
+          "description": "The Approval status for the connection to the Private Link",
+          "type": "string",
+          "readOnly": true,
+          "enum": [
+            "Pending",
+            "Approved",
+            "Rejected",
+            "Disconnected",
+            "Timeout"
```

Comment:
should we limit the set of these enums to just the ones we can validate and keep updated?  to reduce the need for AFD RP to keep needing to get updated status?

Reply:
Are you saying to limit the options to something like the following?
```
Pending
Enabled
Disabled
```
If that's the case, the client will need to query PE to get the full status, and we'll have to maintain the translation between the states that we present to the client and the states we use to communicate with PE and CP. In either approach there is still some level of maintenance required, but I believe this way is a bit more straight-forward.

I'm open to either approach, what do you think?

Reply:
OK, yea, rethink about this, better to keep as it is. Any different set of enum would be breaking change, which makes future updates harder.